SpeechResponder/eddi.json

```diff
@@ -1328,7 +1328,7 @@
                        }\r\n                        {if find(non_stations, type) = -1:\r\n                            {set non_stations to cat(non_stations, [type])}\r\n                        }\r\n                    }\r\n                }\r\n            |elif mission.originsystem = current_system && find(mission.type, \"Delivery\") > -1:\r\n                {if mission.originstation && mission.originstation != \"\":\r\n                    {addstation(mission.originstation)}\r\n                }\r\n            }\r\n        }\r\n    }\r\n    {set stationsCount to len(stations_here)}\r\n\r\n    {if missions_here > 0:\r\n        There {if missions_here > 1: are |else: is } {missions_here}\r\n        {if missions_here = 1: mission |else: missions } {OneOf('here','in this system')}.\r\n\r\n        {if stationsCount  > 0:\r\n            {Pause(500)}\r\n            You have\r\n\r\n            {set cur to 0}\r\n            {while cur < stationsCount:\r\n                {set station to StationDetails(stations_here[cur])}\r\n                {set active to stations_active[cur]}\r\n                {set claim to stations_claim[cur]}\r\n                {set complete to stations_complete[cur]}\r\n                {set depot to stations_depot[cur]}\r\n                {set total to active + claims + complete + depot}\r\n\r\n                {if depot > 0:\r\n                    {depot} depot\r\n                    {if active > 0 && claim > 0: ,\r\n                    |elif active > 0 && complete > 0: ,\r\n                    |elif claim > 0 && complete > 0: ,\r\n                    |elif active > 0 || claim > 0 || complete > 0: and }\r\n                }\r\n                {if active > 0:\r\n                    {active} active\r\n                    {if claim > 0 && completes > 0: ,\r\n                    |elif claim > 0 || complete > 0: and }\r\n                }\r\n                {if claim > 0:\r\n                    {claim} claimable\r\n                    {if complete > 0: and }\r\n                }\r\n                {if complete > 0:\r\n                    {complete} completed\r\n                }\r\n                {if total = 1: mission else: missions} at {P(station.model)}, {station.name}\r\n                {set cur to cur + 1}\r\n                {if cur = stationsCount - 1: and |elif cur < stationsCount: , }\r\n            }.\r\n        }\r\n\r\n        {if missions_here > station_based:\r\n            {Pause(500)}\r\n            {set remaining to missions_here - station_based}\r\n            The {if station_based > 0: remaining {remaining}}\r\n\r\n            {set cur to 0}\r\n            {while cur < len(non_stations):\r\n                {non_stations[cur]}\r\n               \r\n                {set cur to cur + 1}\r\n                {if cur = len(non_stations)-1: and |elif cur < len(non_stations): ,}\r\n            }\r\n            mission{if remaining > 1:s} must be {OneOf(\"searched for\",\"determined\",\"found\")}.\r\n        }\r\n    }\r\n}\r\n",
```

Comment:
Currently, this reads like:
```
There are 2 missions here.  You have 2 active at Or-bis Starport, Newholm Station.
```
It feels a bit redundant / wordy in this case. Consider revising so that the first sentence is only spoken when the number of missions in system differs from the number of missions at stations? Perhaps like:
```
        {if missions_here != station_based:
            There {if missions_here > 1: are |else: is } {missions_here}
            {if missions_here = 1: mission |else: missions } {OneOf('here','in this system')}.
        }

        {if stationsCount  > 0:
            ...
```
This would yield speech like "You have 2 active missions at Newholm Station.", "You have 2 active missions at Newholm Station and 1 active mission at Don's Inheritance.", or "There are 3 missions here.  You have 2 active missions at Or-bis Starport, Newholm Station. The remaining 1 mission must be found."

You are also missing the "|" before the "else" in `{if total = 1: mission else: missions}` so "mission" or "missions" is not verbalized in the final output.

Reply:
To make things easy, this version incorporates the suggestions / fixes recommended above.
```
{_ Mission check system }
{_ Check what missions you have for the current system }


{_ Setup addstation function }
{set addstation(station) to:
    {set missions_here to missions_here + 1}
    {set station_based to station_based + 1}
    {set here to find(stations_here, station)}
    {if here > -1:
        {if mission.statusEDName = "Active":
            {if mission.type = "Collect" ||  mission.type= "Delivery":
                {set depot to stations_depot[here] + 1}
                {set stations_depot to union(stations_depot, [here:depot])}

            |else:
                {set active to stations_active[here] + 1}
                {set stations_active to union(stations_active, [here:active])}
            }
        |elif mission.statusEDName = "Claim":
            {set claim to stations_claim[here] + 1}
            {set stations_claim to union(stations_claim, [here:claim])}
        |elif mission.statusEDName = "Complete":
            {set complete to stations_complete[here] + 1}
            {set stations_complete to union(stations_complete, [here:complete])}
        }
    |else:
        {set stations_here to cat(stations_here, [station])}
        {if mission.statusEDName = "Active":
            {if mission.type = "Collect" ||  mission.type= "Delivery":
                {set stations_active to cat(stations_active, [0])}
                {set stations_depot to cat(stations_depot, [1])}
            |else:
                {set stations_active to cat(stations_active, [1])}
                {set stations_depot to cat(stations_depot, [0])}
            }
            {set stations_claim to cat(stations_claim , [0])}
            {set stations_complete to cat(stations_complete, [0])}
        |elif mission.statusEDName = "Claim":
            {set stations_active to cat(stations_active, [0])}
            {set stations_claim to cat(stations_claim , [1])}
            {set stations_complete to cat(stations_complete, [0])}
            {set stations_depot to cat(stations_depot, [0])}
        |elif mission.statusEDName = "Complete":
            {set stations_active to cat(stations_active, [0])}
            {set stations_claim to cat(stations_claim , [0])}
            {set stations_complete to cat(stations_complete, [1])}
            {set stations_depot to cat(stations_depot, [0])}
        }
    }
}

{if missionsCount > 0:
    {set current_system to system.name}
    {set missions_here to 0}
    {set station_based to 0}
    {set stations_active to []}
    {set stations_claim to []}
    {set stations_complete to []}
    {set stations_depot to []}
    {set stations_here to []}
    {set non_stations to []}

    {_ Find all missions in the current system and identify all the stations }
    {for mission in missions:
        {if mission.status != "Failed":
            {if mission.destinationsystems:
                {for destination in mission.destinationsystems:
                    {if destination.name = current_system && !destination.visited:
                        {set missions_here to missions_here + 1}
                        {if find(mission.typeEDName, "Passenger") > -1:
                            {set type to "Transport "}
                            {set type to cat(type, mission.passengertype)}
                        |else:
                            {set type to mission.type}
                        }

                        {if find(non_stations, type) = -1:
                            {set non_stations to cat(non_stations, [type])}
                        }
                    }
                }
            |elif mission.destinationsystem = current_system:
                {if mission.destinationstation && mission.destinationstation != "":
                    {addstation(mission.destinationstation)}
                |else:
                    {if mission.status = "Active":
                        {set missions_here to missions_here + 1}
                        {if find(mission.typeEDName, "Passenger") > -1:
                            {set type to "Transport "}
                            {set type to cat(type, mission.passengertype)}
                        |else:
                            {set type to mission.type}
                        }
                        {if find(non_stations, type) = -1:
                            {set non_stations to cat(non_stations, [type])}
                        }
                    }
                }
            |elif mission.originsystem = current_system && find(mission.type, "Delivery") > -1:
                {if mission.originstation && mission.originstation != "":
                    {addstation(mission.originstation)}
                }
            }
        }
    }
    {set stationsCount to len(stations_here)}

    {if missions_here > 0:
        {if missions_here != station_based:
            There {if missions_here > 1: are |else: is } {missions_here}
            {if missions_here = 1: mission |else: missions } {OneOf('here','in this system')}.
        }

        {if stationsCount  > 0:
            {Pause(500)}
            You have

            {set cur to 0}
            {while cur < stationsCount:
                {set station to StationDetails(stations_here[cur])}
                {set active to stations_active[cur]}
                {set claim to stations_claim[cur]}
                {set complete to stations_complete[cur]}
                {set depot to stations_depot[cur]}
                {set total to active + claims + complete + depot}

                {if depot > 0:
                    {depot} depot
                    {if active > 0 && claim > 0: ,
                    |elif active > 0 && complete > 0: ,
                    |elif claim > 0 && complete > 0: ,
                    |elif active > 0 || claim > 0 || complete > 0: and }
                }
                {if active > 0:
                    {active} active
                    {if claim > 0 && completes > 0: ,
                    |elif claim > 0 || complete > 0: and }
                }
                {if claim > 0:
                    {claim} claimable
                    {if complete > 0: and }
                }
                {if complete > 0:
                    {complete} completed
                }
                {if total = 1: mission |else: missions} at 
                {Occasionally(2, "the {P(station.model)}, ")}
                {station.name}
                {set cur to cur + 1}
                {if cur = stationsCount - 1: and |elif cur < stationsCount: , }
            }.
        }

        {if missions_here > station_based:
            {Pause(500)}
            {set remaining to missions_here - station_based}
            The {if station_based > 0: remaining {remaining}}

            {set cur to 0}
            {while cur < len(non_stations):
                {non_stations[cur]}
               
                {set cur to cur + 1}
                {if cur = len(non_stations)-1: and |elif cur < len(non_stations): ,}
            }
            {if remaining > 1: missions |else: mission}
            must be {OneOf("searched for","determined","found")}.
        }
    }
}
```

Reply:
For reasons that I have been unable to explain, I also found that the `depot` variable was returning an unexpectedly high number in my test? I had one active depot mission. The value of {depot} was set at 707597 (all other numbers returned the expected count).

Reply:
Incorporated.

Reply:
I haven't seen this issue with {depot} in my testing. I went to great lengths to initialize all variables to zero/empty to insure nothing starts in an unpredictable state.

Can you try to reproduce this? Otherwise, if it's intermittent, I suggest we continue and be mindful of documenting the next occurrence.

Reply:
Ok, I was able to figure out what was going on here. The problem isn't with `{depot}` but rather with `{claims}`. It looks like `{total}` should be using `{claim}` rather than `{claims}`.

There is still a weird `{claims}` variable being set from somewhere which is still returning a very high number.

Reply:
Found it. `{claims}` is being set from the crime monitor.

Reply:
Lol, that's kinda funny... in an aggravating way that only a typo can be.

`total` is only used to state "mission" or "missions". Maybe if you had said that in the first place I could have been more help. ;-)

Good catch & fixed.

Reply:
Incidentally, I don't think `claims` is being set. More likely that the Cottle interpreter assigns `claims` memory location and if you don't initialize it, you get whatever random value is in memory.

This is why I purposefully initialize my variables... found out the hard way. It helps though, when I don't introduce silly typos.

Reply:
I'm pretty sure it is. The number matches the total of the `Claims` showing on the Crime Monitor tab. ;-)